src/transformer.ts

```diff
@@ -0,0 +1,236 @@
+import { ColumnMetadata, TableMetadata } from 'kysely';
+import { AdapterDefinitions, AdapterImports, AdapterTypes } from './adapter';
+import { Definition, DEFINITIONS } from './constants/definitions';
+import { Dialect } from './dialect';
+import { NodeType } from './enums/node-type';
+import { StatementNode } from './nodes';
+import { AliasDeclarationNode } from './nodes/alias-declaration-node';
+import { DeclarationNode } from './nodes/declaration-node';
+import { ExportStatementNode } from './nodes/export-statement-node';
+import { ExpressionNode } from './nodes/expression-node';
+import { GenericExpressionNode } from './nodes/generic-expression-node';
+import { IdentifierNode } from './nodes/identifier-node';
+import { ImportStatementNode } from './nodes/import-statement-node';
+import { InterfaceDeclarationNode } from './nodes/interface-declaration-node';
+import { ObjectExpressionNode } from './nodes/object-expression-node';
+import { PropertyNode } from './nodes/property-node';
+import { UnionExpressionNode } from './nodes/union-expression-node';
+
+const SYMBOLS: { [K in string]?: boolean } = Object.fromEntries(
+  Object.keys(DEFINITIONS).map((key) => [key, false]),
+);
+
+/**
+ * Converts table metadata to a codegen AST.
+ */
+export class Transformer {
+  readonly #declarationNodes: DeclarationNode[];
+  readonly #defaultType: ExpressionNode;
+  readonly #definitions: AdapterDefinitions;
+  readonly #dialect: Dialect;
+  readonly #exportedProperties: PropertyNode[];
+  readonly #imported: Record<string, Set<string>>;
+  readonly #imports: AdapterImports;
+  readonly #symbols = SYMBOLS;
+  readonly #types: AdapterTypes;
+
+  constructor(dialect: Dialect) {
+    this.#declarationNodes = [];
+    this.#defaultType =
+      dialect.adapter.defaultType ?? new IdentifierNode('unknown');
+    this.#definitions = { ...DEFINITIONS, ...dialect.adapter.definitions };
+    this.#dialect = dialect;
+    this.#exportedProperties = [];
+    this.#imported = {};
+    this.#imports = { ColumnType: 'kysely', ...dialect.adapter.imports };
+    this.#symbols = SYMBOLS;
```

Comment:
This should be

```ts
this.#symbols = { ...SYMBOLS };
```

otherwise you end up modifying the global `SYMBOLS` list and any subsequently created instance of `Transformer` will get the previously added symbols (by `#declareSymbol`) by default.

Reply:
Ahh, good catch! Thank you — fixed!